tests/unit/wait.tcl

```diff
@@ -179,33 +179,47 @@ tags {"wait aof network external:skip"} {
                 waitForBgrewriteaof $replica ;# Make sure there is no AOFRW
 
                 $master incr foo
-                assert_equal [$master waitaof 0 1 0] {1 1}
+                assert_equal [$master waitaof 0 1 5000] {1 1}
             }
 
             test {WAITAOF replica copy everysec with AOFRW} {
                 $replica config set appendfsync everysec
 
-                # Need an AOFRW that can be done in a second.
+                # When we trigger an AOFRW, a fsync is triggered when closing the old INCR file,
+                # so with the everysec, we will skip that second of fsync, and in the next second
+                # after that, we will eventually do the fsync.
                 $replica bgrewriteaof
 
                 $master incr foo
-                assert_equal [$master waitaof 0 1 0] {1 1}
+                assert_equal [$master waitaof 0 1 10000] {1 1}
+
+                waitForBgrewriteaof $replica
             }
 
-            test {WAITAOF replica copy everysec->always with AOFRW} {
-                $replica config set appendfsync everysec
+            test {WAITAOF replica copy everysec with slow AOFRW} {
+                $master set foo2 bar
+                $master set foo3 bar
+                wait_for_ofs_sync $master $replica
 
-                waitForBgrewriteaof $replica
+                $replica config set appendfsync everysec
+                $replica config set rdb-key-save-delay 1000000 ;# 1 sec
 
-                # Need an AOFRW that can be done in a second.
+                # There are three keys in total, so AOFRW takes three seconds to complete.
                 $replica bgrewriteaof
 
                 $master incr foo
+                assert_equal [$master waitaof 0 1 10000] {1 1}
 
-                # Change appendfsync from everysec to always
-                $replica config set appendfsync always
+                $replica config set rdb-key-save-delay 0
+                waitForBgrewriteaof $replica
+            }
 
-                assert_equal [$master waitaof 0 1 0] {1 1}
+            test {WAITAOF replica copy everysec->always with AOFRW} {
+                $replica config set appendfsync everysec
+                $replica bgrewriteaof
+                $master incr foo
```

Comment:
there's a race here, we don't know who's gonna finish first.
let's make it predictable.

Reply:
In the current code, no matter which one finishes first, it should not cause hang (this is our goal)
i write in on purpose, do you have some ideas to make it predictable?

Reply:
adding either a waitForBgrewriteaof or rdb-key-save-delay will make it predictable.

Reply:
i want to make a fast AOFRW. like
- bgrewriteaof (fsync and close the old aof)
- INCR
- the end of AOFRW
- everysec to always
- 
These are all done in the same second

Reply:
ok, so you want the rewrite to end in less than a second (still it will not always be in the same, round second)
do you want to make sure the WAITAOF ends before changing the config?

if we want to try to fit all of them in the same round second (although there's no way to guarantee that, let's document it.

Reply:
updated it